packages/xo-server-transport-icinga2/src/index.js

```diff
@@ -0,0 +1,136 @@
+import assert from 'assert'
+import createLogger from '@xen-orchestra/log'
+import httpRequestPlus from 'http-request-plus'
+import { URL } from 'url'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:transport-icinga2')
+
+// =============================================================================
+
+export const configurationSchema = {
+  type: 'object',
+
+  properties: {
+    server: {
+      type: 'string',
+      description: `
+The icinga2 server http/https address.
+
+*If no port is provided in the URL, 5665 will be used.*
+
+Examples:
+- https://icinga2.example.com
+- http://192.168.0.1:1234
+      `.trim(),
+    },
+    user: {
+      type: 'string',
+      description: 'The icinga2 server username',
+    },
+    password: {
+      type: 'string',
+      description: 'The icinga2 server password',
+    },
+    filter: {
+      type: 'string',
+      description: 'The filter to use',
+    },
+    acceptUnauthorized: {
+      type: 'boolean',
+      description: 'Accept unauthorized certificates',
+      default: false,
+    },
+  },
+  additionalProperties: false,
+  required: ['server', 'filter'],
+}
+
+// =============================================================================
+
+const STATUS_MAP = {
+  OK: 0,
+  WARNING: 1,
+  CRITICAL: 2,
+  UNKNOWN: 3,
+}
+
+// =============================================================================
+
+class XoServerIcinga2 {
+  constructor({ xo }) {
+    this._xo = xo
+  }
+
+  // ---------------------------------------------------------------------------
+
+  configure(configuration) {
+    const serverUrl = new URL(configuration.server)
+    if (configuration.user !== '') {
+      serverUrl.username = configuration.user
+    }
+    if (configuration.password !== '') {
+      serverUrl.password = configuration.password
+    }
+    if (serverUrl.port === '') {
+      serverUrl.port = '5665' // Default icinga2 access port
+    }
+    serverUrl.pathname = '/v1/actions/process-check-result'
+    this._url = serverUrl.href
+
+    this._filter = configuration.filter
+    this._acceptUnauthorized = configuration.acceptUnauthorized
+  }
+
+  load() {
+    this._unset = this._xo.defineProperty(
+      'sendIcinga2Status',
+      this._sendIcinga2Status,
+      this
+    )
+  }
+
+  unload() {
+    this._unset()
+  }
+
+  test() {
+    return this._sendIcinga2Status({
+      message:
+        'The server-icinga2 plugin for Xen Orchestra server seems to be working fine, nicely done :)',
+      status: 'OK',
+    })
+  }
+
+  // ---------------------------------------------------------------------------
+
+  async _sendIcinga2Status({ message, status }) {
+    const icinga2Status = STATUS_MAP[status]
+    assert(icinga2Status !== undefined, `Invalid icinga2 status: ${status}`)
+    try {
+      await httpRequestPlus
```

Comment:
You can use [`this._xo.httpRequest`](https://github.com/vatesfr/xen-orchestra/blob/fd063743658b04aeeaddc46fa7fc05ce47a24b45/packages/xo-server/src/xo-mixins/http.js#L14).

Reply:
Indeed, this may be better because it will use `xo-server` HTTP proxy config.